### DISCUSSION TITLE
"Maximum Number of Briefcases Per User Reached" Error When Syncing File to iModel Hub

### DISCUSSION THREAD

**Discussion by VietBuiHa** [iTwin/community#60] [Q&A] [answered]
Hello,

I am encountering an issue while trying to sync a file to iModel Hub. The process fails with the following error message:

`WSClient - WSError: Maximum number of briefcases per user reached. (ActivityId: '0caeb132-a4a2-4548-b815-01d6416caa4e')`

From my understanding, it seems that the limit for the number of briefcases associated with my user account has been exceeded. However, I haven't found a way to resolve this or manage existing briefcases effectively.

Questions:

What is the maximum number of briefcases allowed per user, and is there a way to increase this limit?
How can I manage or delete existing briefcases to free up space for new syncs?
Is there a recommended best practice for handling briefcases when dealing with frequent updates and syncs?
I appreciate any guidance or solutions you can provide to help resolve this issue.

Thank you!

Enclosed is the log file

[2f1839e1-85b8-429e-9eb9-3cb2a995ebd8.log](https://github.com/user-attachments/files/17797242/2f1839e1-85b8-429e-9eb9-3cb2a995ebd8.log)

**Comment by robertas-kerpys** [comment, marked as answer]
Hello, 

thanks for reaching out!

> _What is the maximum number of briefcases allowed per user, and is there a way to increase this limit?_

You are right, you have reached a [limit of the briefcases](https://developer.bentley.com/apis/imodels-v2/operations/acquire-imodel-briefcase/) per user. Current limit is 100 briefcases per user. 
[BriefcaseId](https://www.itwinjs.org/learning/imodelhub/briefcases/#briefcaseid)'s are a limited resource thus they need limits per user to not grow uncontrollably. 

> _How can I manage or delete existing briefcases to free up space for new syncs?_

You need to release the briefcases which you are not using. Please leverage [Release iModel Briefcase](https://developer.bentley.com/apis/imodels-v2/operations/release-imodel-briefcase/) API to release unused briefcases.

> _Is there a recommended best practice for handling briefcases when dealing with frequent updates and syncs?_

Yes, the guidance is to reuse the acquired briefcase. That is, you pull the changes from the iModels Service (iModelHub) and push your changes to iModels Service. The next time you run the sync, you reuse the same briefcase acquired earlier to pull and push the changes. 

From the error message and the logs I see that you are using a deprecated API and SAML authorization. Please prioritize upgrading your application to use the new API - [iModels API](https://developer.bentley.com/apis/imodels-v2/overview/).

Let me know if you need any other information on this issue.

**Comment by VietBuiHa** [comment]
Hello Robert,

Thank you for your prompt and detailed response!

After clearing all briefcases, our syncs are now functioning normally again, which is great news. Your guidance on managing and reusing briefcases has been very helpful.

Regarding the deprecated API and SAML authorization, we’re a bit unsure about this. We're using the iTwin Synchronizer to sync our files and are confident that we’ve already updated it to the latest version for our connectors. Could you clarify if there might be any additional steps or configurations we need to check to ensure full compliance with the latest iModels API?

**Comment by robertas-kerpys** [comment]
Which version of iTwin Synchronizer are you leveraging? iTwin Synchronizer version 1.1.506 and above uses new iModels API.
